### ISSUE TITLE
Support for more asynchronous chat

### ISSUE THREAD

**Issue by micahflee** [onionshare/onionshare#1296]
I received some feedback about chat. Right now if you start a chat server and then wait for someone to join, and your screen locks, the websocket connection in Tor Browser disconnects. So if you later unlock your screen and connect again, you miss anything that happened while you were gone.

I wonder how we can improve this, at least for the host who's running OnionShare. @SaptakS

**Comment by SaptakS** [comment]
Yes. This has been an issue I am aware of. Tor drops the websocket connections if there has been no activity for a long time. Given that the socket server in the onionshare server side is still running, the only way probably would be to detect that the client is no longer connected to the websocket, and then retry connecting.

Theoretically that should be possible. There are 2 main steps (and both need some looking into):

1. A good way to detect you are disconnected (i.e. detect that it is dropped via tor and not the actual chat server stopped)
2. Once detected, try to reconnect to the server again (not sure if this will actually be as easy as making a reconnect or a new connect from JS, or need some other strategy)

This is clearly a tor/tor brrowser phenomenon (not sure about the intricacies), because if a server is started locally and visited by local link, this issue doesn't appear.
